homeassistant/components/logbook/models.py

```diff
@@ -123,7 +125,7 @@ def async_event_to_row(event: Event) -> EventAsRow:
         return EventAsRow(
             data=event.data,
             context=event.context,
-            event_type=event.event_type,
+            event_type=str(event.event_type),
```

Comment:
Should we update the `event_type` attribute typing for class `EventAsRow`?

Reply:
How can this not be a string?

Reply:
> How can this not be a string?

The idea of this PR is to add a new type (`EventType`) that is generic and behaves similar to a string without actually being one. That can subsequently be used to improve type checking `async_fire` and `async_listen` calls.

Reply:
I was a bit worried about that but it looks like we don't actually check the event type very often in the core code since we are usually listening only for one type.

I wonder how this will behave with templates

Reply:
If `orjson` serializes these natively without having to fallback to the default handler it would be better to change the type and not do the conversion here.

If it doesn't than its better to convert it here

Reply:
```
>>> from homeassistant.const import EVENT_STATE_CHANGED
>>> import orjson
>>> orjson.dumps({"event_type":EVENT_STATE_CHANGED})
b'{"event_type":{"name":"state_changed"}}'
```